framework/auth/core.py

```diff
@@ -280,20 +319,27 @@ class User(GuidStoredObject, AddonModelMixin):
     # }
 
     # Time of last sent notification email to newly added contributors
+    contributor_added_email_records = fields.DictionaryField(default=dict)
     # Format : {
     #   <project_id>: {
     #       'last_sent': time.time()
     #   }
     #   ...
     # }
-    contributor_added_email_records = fields.DictionaryField(default=dict)
 
     # The user into which this account was merged
     merged_by = fields.ForeignField('user', default=None, index=True)
 
-    # verification key used for resetting password
+    # verification key v1,
     verification_key = fields.StringField()
 
+    # verification key v2, with expiration time and one-time only
+    verification_key_v2 = fields.DictionaryField(default=dict)
```

Comment:
For CR: do I need to add migration to update each user with new fields for this and below? or adding checks that catch `AttributeError` or `KeyError` (and consider them as invalid or expired) is better?